diff -Naurp kodi/xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemuxFFmpeg.cpp kodi-patch/xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemuxFFmpeg.cpp
--- kodi/xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemuxFFmpeg.cpp	2023-11-01 15:48:40.969887774 +0000
+++ kodi-patch/xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemuxFFmpeg.cpp	2023-11-05 15:08:39.000000000 +0000
@@ -208,6 +208,9 @@ CDVDDemuxFFmpeg::CDVDDemuxFFmpeg() : CDV
   m_bSup = false;
   m_speed = DVD_PLAYSPEED_NORMAL;
   m_program = UINT_MAX;
+  m_oldStreams = 0;
+  m_oldProgCount = 0;
+  m_bResetStream = false;
   m_pkt.result = -1;
   memset(&m_pkt.pkt, 0, sizeof(AVPacket));
   m_streaminfo = true; /* set to true if we want to look for streams before playback */
@@ -600,6 +603,15 @@ bool CDVDDemuxFFmpeg::Open(const std::sh
     skipCreateStreams = true;
   }
 
+  if (m_bResetStream)
+  {
+    CLog::Log(LOGINFO, "CDVDDemuxFFmpeg::Open(): Reset Stream");
+    m_bResetStream = false;
+    m_program = 0;
+    m_checkTransportStream = true;
+    skipCreateStreams = true;
+  }
+
   // reset any timeout
   m_timeout.SetInfinite();
 
@@ -1091,14 +1103,39 @@ DemuxPacket* CDVDDemuxFFmpeg::ReadIntern
           av_dump_format(m_pFormatContext, 0, CURL::GetRedacted(m_pInput->GetFileName()).c_str(),
                          0);
 
-          // update streams
-          CreateStreams(m_program);
+          unsigned int m_newStreams = m_pFormatContext->nb_streams;
+          if (m_oldStreams == 0)
+            m_oldStreams = m_newStreams;
+          CLog::Log(LOGINFO, "CDVDDemuxFFmpeg::Read(): Old streams - {} / New streams - {}", m_oldStreams, m_newStreams);
+
+          unsigned int m_newProgCount = m_pFormatContext->nb_programs;
+          if (m_oldProgCount == 0)
+            m_oldProgCount = m_newProgCount;
+          CLog::Log(LOGINFO, "CDVDDemuxFFmpeg::Read(): Old programs - {} / New programs - {}", m_oldProgCount, m_newProgCount);
 
-          pPacket = CDVDDemuxUtils::AllocateDemuxPacket(0);
-          pPacket->iStreamId = DMX_SPECIALID_STREAMCHANGE;
-          pPacket->demuxerId = m_demuxerId;
+          if (m_newStreams > m_oldStreams+1 || m_newProgCount > m_oldProgCount)
+          {
+            // reset streams
+            m_oldStreams = 0;
+            m_oldProgCount = 0;
+            m_bResetStream = true;
+            Reset();
+            return nullptr;
+          }
+          else
+          {
+            m_oldStreams = m_newStreams;
+            m_oldProgCount = m_newProgCount;
 
-          return pPacket;
+            // update streams
+            CreateStreams(m_program);
+
+            pPacket = CDVDDemuxUtils::AllocateDemuxPacket(0);
+            pPacket->iStreamId = DMX_SPECIALID_STREAMCHANGE;
+            pPacket->demuxerId = m_demuxerId;
+
+            return pPacket;
+          }
         }
 
         AVStream* stream = m_pFormatContext->streams[m_pkt.pkt.stream_index];
diff -Naurp kodi/xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemuxFFmpeg.h kodi-patch/xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemuxFFmpeg.h
--- kodi/xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemuxFFmpeg.h	2023-10-29 22:31:26.000000000 +0000
+++ kodi-patch/xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemuxFFmpeg.h	2023-11-02 10:46:38.000000000 +0000
@@ -152,7 +152,10 @@ protected:
   bool     m_bMatroska;
   bool     m_bAVI;
   bool     m_bSup;
+  bool     m_bResetStream;
   int      m_speed;
+  unsigned int m_oldStreams;
+  unsigned int m_oldProgCount;
   unsigned int m_program;
   unsigned int m_streamsInProgram;
   unsigned int m_newProgram;
